sql/planbuilder/builder.go

```diff
@@ -184,6 +199,13 @@ func (b *Builder) Reset() {
 	b.viewCtx = nil
 	b.nesting = 0
 	b.qFlags = &sql.QueryFlags{}
+	if b.authHandler != nil {
+		var err error
+		b.queryHandler, err = b.authHandler.NewQueryHandler(b.ctx, b.cat)
```

Comment:
do we really need a new object for each query?

Reply:
Makes it easier to keep the state needed for that query. I could have it in a struct that's passed to the interface on each call, but they seem equivalent.

Reply:
the state here is `ctx`?

Reply:
`ctx` along with the user and privilege set. Rather than grabbing it for each check, we grab it once and reuse it.